app/scripts/lib/errors.js

```diff
@@ -253,8 +253,10 @@ define(function (require, exports, module) {
     normalizeXHRError: function (xhr) {
       var err;
 
-      if (! xhr || xhr.status === 0) {
+      if (! xhr || xhr.status === 0 || xhr.status === 503 ) {
```

Comment:
Can you ensure that both 503 and 429's are normalized IFF there is no xhr.responseJSON? Otherwise, we are going to lose all of the response JSON body which provides valuable information when available.